retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.ts

```diff
@@ -0,0 +1,181 @@
+import { PageEvent } from '@angular/material/paginator';
+import { MatDialog } from '@angular/material/dialog';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { EmployeeRepository } from 'src/app/domain/employee/employee.repository';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { TranslateService } from '@ngx-translate/core';
+import { ConfirmDialogComponent } from 'src/app/shared/components/confirm-dialog/confirm-dialog.component';
+import { Employee } from 'src/app/domain/employee/models/employee';
+import { map } from 'rxjs';
+import { Pagination } from 'src/app/core/models/pagination';
+@Component({
+  selector: 'app-employee',
+  templateUrl: './employee.component.html',
+  styleUrls: ['./employee.component.scss']
+})
+export class EmployeeComponent implements OnInit {
+  dataSource: Employee[] = [];
+  paginationEmployee!: Pagination
+  pageEvent!: PageEvent;
+  employeeForm!: FormGroup;
+  submit: boolean = false;
+  currentData!: Employee;
+  isButtonVisible: boolean = true;
+  displayedColumns: string[] = [
+    'id',
+    'fullName',
+    'nickName',
+    'nationalId',
+    'primaryPhoneNo',
+    'secondaryPhoneNo',
+    'address',
+    'job',
+    'actions',
+  ];
+  constructor(
+    private employeesRepository: EmployeeRepository,
+    private build: FormBuilder,
+    private _snackBar: MatSnackBar,
+    private dialog: MatDialog,
+    private translate: TranslateService
+  ) { }
+
+  ngOnInit() {
+    this.employeeDataForm();
+    this.getAllEmployee()
+    this.pagination()
+  }
+  getAllEmployee() {
+    this.employeesRepository.getList().subscribe((result) => {
+      this.dataSource = result.data;
+    });
+  }
+
+  pagination() {
+    this.employeesRepository.getList().subscribe((result) => {
+      this.paginationEmployee = result.pagination;
+    });
+  }
+  onPaginationChange(event: PageEvent) {
+    let page = event.pageIndex;
+    let size = event.pageSize;
+    this.employeesRepository.findAll(page, size).pipe(
```

Comment:
you already have function that get employee why make another?
call function getAllEmployee()